scripts/list-changes-since.py

```diff
@@ -35,28 +36,70 @@ def get_last_tag():
     return tags.stdout.decode().splitlines()[0]
 
 
-def get_prs_closed_since(version):
-    git_list_result = run(['git','log','--oneline','{}..master'.format(version)],
+def get_tag_date(tag):
+    tag_date_result = run(['git', 'tag', '--list', tag, '--format=%(taggerdate)'],
         stdout=PIPE)
-    prs = []
-    for message in git_list_result.stdout.split(b'\n'):
-        match = re.match('.*Merge pull request #([0-9]+)', str(message))
-        if match:
-            prs.append(int(match.group(1)))
-    return prs
+    tag_date_str = tag_date_result.stdout.decode().strip()
+    return dateutil.parser.parse(tag_date_str)
+
 
+def github_request(auth_token, repo, path, **kwargs):
+    """
+    Make a GitHub API request and return an iterator over items in the response.
 
-def get_pr_info(auth_token, repo, id):
-    pr_url = '{}/repos/{}/pulls/{}'.format(GITHUB_API_URL, repo, id)
+    `github_request` follows `next` links in paged responses automatically.
+    """
+    params = kwargs
+    url = '{}/repos/{}/{}'.format(GITHUB_API_URL, repo, path)
     headers = {}
     if auth_token:
         headers['Authorization'] = 'token {}'.format(auth_token)
-    res = requests.get(pr_url, headers=headers)
-    if res.status_code != 200:
-        raise Exception('GitHub request failed:', res.json()['message'])
-    data = res.json()
 
-    return PRInfo(id, title=data['title'].strip())
+    while url:
+        res = requests.get(url, headers=headers, params=params)
+        if res.status_code != 200:
+            raise Exception('GitHub request failed:', res.text)
+
+        page = res.json()
+        if isinstance(page, list):
+            for item in page:
+                yield item
+            url = res.links.get('next') and res.links.get('next')['url']
```

Comment:
FWIW this kind of thing isn't really idiomatic in Python, IMO. Probably something more like:

```python
try:
    url = res.links['next']['url']
except KeyError:
    url = None
```